### ISSUE TITLE
Cannot generate blast files

### ISSUE THREAD

**Issue by Wooba** [mjsull/Easyfig#12]
Windows 7, 64 bit computer, Experience with computers: could not write a line of code to safe my life.

I wish to compare two sequences about 57 Kb in size for my paper.
I have exported my annotated files from CLC and saved them as both .gbk and .fa
I downloaded the EasyFig 2.2.2 and extracted it into a folder in My Documents
The program can run and I can easily add my sequences by pressing the [Add feature file] button.
When I press [Generate blastn Files] or [Download BLAST automatically] it starts downloading BLAST, but gets stuck after download is complete with the message "Downloading ...Complete, Extracting file...". After a while the black window reads:

![easyfig1](https://user-images.githubusercontent.com/35774567/35344306-1e0aaf9c-012d-11e8-8188-93cad8b04610.png)

So I decided to download BLAST manually, the program suggests I download ncbi-blast-2.7.1+-win64.exe. I install it in the default C:\Program Files\NCBI
I install BLAST and restart the EasyFig application. Again I load in the files and goes into [Blast]->[Choose BLAST Location], I find the C:\Program Files\NCBI\blast-2.7.1+\bin
Now I'm ready to generate so I press [Generate blastn Files], I now have to choose a location. I select same folder as where my sequences are in My Documents
![easyfig2](https://user-images.githubusercontent.com/35774567/35344882-735d2ab4-012e-11e8-8579-267e92d4ff02.png)

Then I press save as and choose my folder in My Documents again
After I name the file an error pops up [A selected file does not exist [OK] ]

I know my folder names may contain spaces or my paths are pretty long. A new version 2.2.3 is added but I don't know what to do with a .py file

I believe a tutorial where we have to generate the BLAST files ourselves would be a very nice addition to the wiki

Help is much appreciated

**Comment by plabreck** [comment]
Same problem. Have you figured out a solution?

**Comment by felipelira** [comment]
Maybe I am not alone with this problem. Nothing for the moment? Using Linux it is ok but for windows don't.

**Comment by mjsull** [comment]
Hi all,

There's a new version for windows that will hopefully fix this issue. Please let me know if it worked.

Available here:
http://mjsull.github.io/Easyfig/files.html

-Mitch

**Comment by felipelira** [comment]
I am using the w.w.3 version since yesterday and doesn't work.

Felipe Lira, PhD
Bioinformatician

Before you print this e-mail, think well if it is really necessary. 

    En lunes, 27 de agosto de 2018 1:51:32 CEST, mjsull <notifications@github.com> escribió:  
 
 
Hi all,

There's a new version for windows that will hopefully fix this issue. Please let me know if it worked.

Available here:
http://mjsull.github.io/Easyfig/files.html

-Mitch

**Comment by plabreck** [comment]
I just tried the version 2.2.3 for windows and was unable to generate the blastn files. It looks as if they are made but they do not appear in the designated folder.... The only files that appear are the input fasta files needed to make the blastn output. 

The Blast files in the image are unreachable

Thanks,
Pat

![pic1](https://user-images.githubusercontent.com/42583550/44653861-e3c43580-a9bd-11e8-9918-df35db5ced59.png)

**Comment by mjsull** [comment]
I think I found the problem.

Does someone mind telling me if this fixes it.
[Download](https://github.com/mjsull/Easyfig/releases/download/2.2.4/Easyfig_2.2.4_win.zip)

Cheers,

Mitch

**Comment by plabreck** [comment]
Mitch,
I gave version 4 a try and still the same result. The blastn files are not
being saved or I am unable to find them. I am running the software with
windows 7.


Best,
Pat

**Comment by mjsull** [comment]
Hi Pat,

After extracting the folder did you use the Download BLAST+ automatically too?

-Mitch

**Comment by plabreck** [comment]
Mitch,
I've tried it with the BLAST+ program I had downloaded previously and the
automatically downloaded BLAST+ feature. Same result-- "A selected file
does not exist"

Pat

**Comment by felipelira** [comment]
The same for me, just in case. I moved to Linux and it works nice.

**Comment by mjsull** [comment]
Could someone try a couple of things for me.

Can you tell me if you see

libgcc_s_seh-1.dll
libgmp-10.dll
libgnutls-30.dll
libhogweed-4-2.dll
libnettle-6-2.dll
libp11-kit-0.dll
msvcp120.dll
msvcr120.dll
makeblastdb.exe
blastn.exe

in the folder where the Easyfig executable is?

Also if you go into "About your PC" under "Device specifications"
could you tell me what the "System type" is

and under "Windows specifications"
what "edition" you are using.

Thanks,

Mitchell

**Comment by plabreck** [comment]
Hi Mitch,
All the files are present in Easyfig 2.4 in windows. They are in subfolders
separate from the easyfig executable but everything is contained within the
easyfig folder itself.

Windows 7 Professional
64 bit I7 @2.5GHZ
8GB Ram

Best,
Pat

**Comment by ammaraziz** [comment]
Hi, 

I had a similar problem as described by @plabreck and was getting  an error "a file selected was not found". 

Version 2.2.4 solved the issue.

I'm on Windwos 10, 64 bit. I let EasyFig to download and install blast in both version. I noticed in version 2.2.3 blast would finish in seconds. But in 2.2.4 it took a few minutes and produced correct outputs. 

I think blast is not even executing in version 2.2.3. 

Thanks!

**Comment by thsyd** [comment]
Hi,
I'm having same problems as described above. Tried with the 2.2.4 version downloaded from the link 
https://github.com/mjsull/Easyfig/releases/download/2.2.4/Easyfig_2.2.4_win.zip.

.fa files are generated, but that's it.

OS Name	Microsoft Windows 10 Enterprise
Version	10.0.17134 Build 17134
Processor	Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz, 2712 Mhz, 2 Core(s), 4 Logical Processor(s)

Blast+2.7.1 installed C:\Program Files\NCBI\blast-2.7.1+\bin

I do not have 
libgcc_s_seh-1.dll
libgmp-10.dll
libgnutls-30.dll
libhogweed-4-2.dll
libnettle-6-2.dll
libp11-kit-0.dll
msvcp120.dll
msvcr120.dll
makeblastdb.exe
blastn.exe

in the folder where the Easyfig.exe is (only python27.dll and three folders lib, tcl and tk).


EDIT: and the Easyfig.exe version in the Easyfig_2.2.4_win.zip, says 2.2.3 on the easyfig windows and in the "about" menu item. 

EDIT: So I tried running Easyfig.py on WSL ubuntu using VcXsrv and thereby running GUI applications.
When I try to Generate blastn Files the .out files are not generated. As I've launced the Easyfig.py from the WSL terminal I can see some output on the WSL termina. Here I can see the Blast+ (2.2.31+) error message: "Error: Too many positional arguments (1), the offending value: -

**Comment by brojobezares** [comment]
Hi, 
I have the same problem as described above. I try to download blast automatically, but the programs finished in .dll not download. I can see the programs makeblastdb.exe and blastn.exe. 
If I try to download blast manually, what version I should have to download for Windows 7 Professional 64 bit 8GB Ram. 
Directory listing
Parent Directory (Root Directory)
      85 Nov 26 21:10 ChangeLog
20103201 Nov 26 21:10 ncbi-blast-2.8.1+-2.src.rpm
      62 Nov 26 21:13 ncbi-blast-2.8.1+-2.src.rpm.md5
194792784 Nov 26 21:10 ncbi-blast-2.8.1+-2.x86_64.rpm
      65 Nov 26 21:13 ncbi-blast-2.8.1+-2.x86_64.rpm.md5
24973554 Nov 26 21:13 ncbi-blast-2.8.1+-src.tar.gz
      63 Nov 26 21:13 ncbi-blast-2.8.1+-src.tar.gz.md5
28874837 Nov 26 21:13 ncbi-blast-2.8.1+-src.zip
      60 Nov 26 21:13 ncbi-blast-2.8.1+-src.zip.md5
94442276 Nov 26 21:10 ncbi-blast-2.8.1+-win64.exe
      62 Nov 26 21:13 ncbi-blast-2.8.1+-win64.exe.md5
241992963 Nov 26 21:12 ncbi-blast-2.8.1+-x64-linux.tar.gz
      69 Nov 26 21:13 ncbi-blast-2.8.1+-x64-linux.tar.gz.md5
134348474 Nov 26 21:09 ncbi-blast-2.8.1+-x64-macosx.tar.gz
      70 Nov 26 21:13 ncbi-blast-2.8.1+-x64-macosx.tar.gz.md5
94159028 Nov 26 21:10 ncbi-blast-2.8.1+-x64-win64.tar.gz
      69 Nov 26 21:13 ncbi-blast-2.8.1+-x64-win64.tar.gz.md5
136049938 Nov 26 21:08 ncbi-blast-2.8.1+.dmg
      56 Nov 26 21:13 ncbi-blast-2.8.1+.dmg.md5

Kind regards, 
Beatriz

**Comment by ipsalto** [comment]
Hi! I´m having the same problem. I downloaded the 2.2.4 version from your link (https://github.com/mjsull/Easyfig/releases/download/2.2.4/Easyfig_2.2.4_win.zip) and the program still  does not generate the blast files.
I do not have the followings (either in the executable folder, neither  in three folders lib, tcl and tk)

libgcc_s_seh-1.dll
libgmp-10.dll
libgnutls-30.dll
libhogweed-4-2.dll
libnettle-6-2.dll
libp11-kit-0.dll
msvcp120.dll
msvcr120.dll


I only have
 python27.dll
makeblastdb.exe
blastn.exe
tblastx.exe


Windows 7 Professional
64 bits
8GB RAM

Did you solve this problem?

Kind regards

ileana

**Comment by mjsull** [comment]
Hi,

https://raw.githubusercontent.com/mjsull/Easyfig/master/debug_blast_dl.py

If someone with python installed could download the above and run 

"python /path/to/download/debug_blast_dl.py"

Hopefully I'll be able to fix this.

-Mitch

**Comment by murd0ck-pb** [comment]
Output is:

Windows
64bit
True platform not found

**Comment by thsyd** [comment]
on WSL / windows 10.

```
$ python debug_blast_dl.py
Linux
64bit
True platform not found 
```

**Comment by Cedric-woudstra** [comment]
I had the same problem and solved it by running easyfig in admin mode

**Comment by thsyd** [comment]
Hi. Now it works for me as well by running in Windows 10 - run as administrator.
Also, I use a data folder just above the c: drive (eg c:/test/) with no spaces. I don't know if this actually helps anything.

**Comment by brittbennett** [comment]
Hi Mitch. I have tried running in Windows 10 as administrator, but I still get the same error message others have gotten ("a selected file does not exist") when I try to create a figure. When I run "Generate blastn files" it only returns easyfig.fa files to my destination folder, even though it says it's generated a .out file in the program. I assume this is the file that does not exist?

I also noticed that, even though my download folder says 2.2.4, the version I get when I open the program is still 2.2.3. Could this have something to do with my difficulty generating a blastn file? Is there something else I should be doing that I've missed from your earlier replies?

Thank you,
Britt

![easyfig blastn](https://user-images.githubusercontent.com/27703167/62406976-9f85d180-b54c-11e9-984a-368e4be113c1.jpg)

**Comment by DSPINTO21** [comment]
Hello,
I've had the exact same issue as mentioned above where once "generate blastn" is clicked the .out file appears but in its location appear two .fa files.

I am currently on version 2.2.5 on Windows 10 x64 bit.

Any help would be much appreciated 

Thank You,
Pinto

**Comment by lngia1** [comment]
Hi there,

I've experienced similar problem where the blastn file output appears to have the FASTA file only, and previously I've keep receiving error message "A selected file does not exist"

However, I've figured the problem might be with the space in your filename or path directory. It appears the file pathway or filename cannot have any space. So maybe stick to rename your folder name and file name with no space. Also I've run it on admin mode. 

Now there's no more error message appearing and image appears fairly quick too. 

Hope that helps
Lyman

**Comment by DSPINTO21** [comment]
hiya,
I've done that aswell, I made sure non of the file/folders in the path directory had any spaces, but I still keep getting the same error. I did it on an ubuntu system and it worked perfectly.

Thank You
Pinto

**Comment by DrGin-hub** [comment]
Hi All, I faced the same issue and I solved it. I noticed that my mistake was to select a different working directory, when I clicked on generate tblatx /blastn, seems that binaries makeblastdb, blastn and tblastx are setup in the folder Easyfig_VERSION that you download. Check if the binaries are there or search in your system for them. Another issue was related to the executable libz.so.1 inside of the easyfig I did a symlink to solve it : 

cd /(the easyfig directory containing libz.so.1)
mv libz.so.1 libz.so.1.notused
ln -s /lib/x86_64-linux-gnu/libz.so.1

Tested on Ubuntu 20.04
Easyfig 2.2.2
